z_infra_provisioning/cloud_infra_center/ocp_upi/roles/configure-pre-check/tasks/main.yaml

```diff
@@ -87,4 +87,23 @@
 - name: Check server group quota
   fail:
     msg: "Server group number is not enough, current: {{ server_group.stdout_lines[0] }}, required: 1"
-  failed_when: "server_group.stdout_lines[0]|int < 1"
\ No newline at end of file
+  failed_when: "server_group.stdout_lines[0]|int < 1"
+
+- name: Get storage provide
+  shell: openstack volume service list -f value | grep cinder-volume |awk '{print$4$5}'
+  register: storage_backend
+  when: use_storage_provider == true
+
+- name: Check if storage provide is configured correctly
+  fail:
+    msg: "Storage provider is not enabled and active."
+  failed_when: 
+  - storage_backend.stdout_lines[0] != 'enabledup'
```

Comment:
when there are multiple storage provider registered in an ICIC, this would always check the first Storage provider?
You need to check the storage provider corresponding to the user-specified volume type.